src/kernelspace/exec_monitor.c

```diff
@@ -25,14 +25,42 @@ struct {
 } ringbuf SEC(".maps");
 
 long ringbuffer_flags = 0;
+int max_n_proc;
+int n_monitored_proc;
+
+int filter_by_ppid;
+int ppid_list[10];
+int ppid_list_size;
 
 SEC("lsm/bprm_committed_creds")
 void BPF_PROG(exec_audit, struct linux_binprm *bprm)
 {
+	if (max_n_proc != 0 && n_monitored_proc == max_n_proc)
```

Comment:
Variable declaration should be before any code in the coding style you use

Reply:
Note that if you move the count to the userspace this and the variables it checks is no longer needed